File: module4/01_Spring_MVC_Overview/exercise/dictionary/src/main/java/com/service/DictionaryService.java
package com.service;

import org.springframework.stereotype.Service;

import java.util.HashMap;
import java.util.Map;
@Service
public class DictionaryService implements IDictionaryService {
    static Map<String,String> map = new HashMap<>();
    static {
        map.put("banana","chuối");
        map.put("cat","mèo");
        map.put("dog","chó");
    }
    @Override
    public String getMeaning(String key) {
        return map.get(key)!=null?map.get(key):"the word you entered is not in the dictionary";
    }
}

File: module5/07_angular_service_router/exercise/dictionary-lookup-app/src/app/service/dictionary.service.ts
import { Injectable } from '@angular/core';
import {IWord} from '../i-word';

@Injectable({
  providedIn: 'root'
})
export class DictionaryService {
  dictionary: IWord[] = [
    {word: 'cat', mean: 'mèo'},
    {word: 'dog', mean: 'chó'},
    {word: 'banana', mean: 'chuối'},
    {word: 'ayame', mean: 'oni'}
    ];

  constructor() { }
  translate(word: string): IWord {
   return  this.dictionary.find(value => value.word === word);
  }
  getAll(): IWord[] {
    return this.dictionary;
  }
  findWord(keyword: string): IWord[] {
    return this.dictionary.filter(value => value.word.includes(keyword));
  }
}
